src/lib/operators/abstract_operator.cpp

```diff
@@ -168,14 +164,16 @@ void AbstractOperator::execute() {
 }
 
 std::shared_ptr<const Table> AbstractOperator::get_output() const {
-  Assert(_executed, "Trying to get_output of operator that was not executed yet.");
+  Assert(executed(), "Trying to get_output of operator that was not executed yet.");
   return _output;
 }
 
 void AbstractOperator::clear_output() {
-  Assert(_executed, "Unexpected call of clear_output() since operator did not execute yet.");
   Assert(_consumer_count == 0, "Cannot clear output since there are still consuming operators.");
-  if (!_never_clear_output) _output = nullptr;
+  if (!_never_clear_output) {
+    _transition_to(OperatorState::Cleared);
```

Comment:
Did you consider to always perform the transition? I feel like that might help us in catching additional bugs.

Reply:
I did this on purpose. An Assert failed because of the transition from `OperatorState::Cleared` to `OperatorState::Cleared`. 
Since `clear_output` is idempotent, I did not want to introduce any other locks and allowed for this.

Reply:
>  Since clear_output is idempotent, I did not want to introduce any other locks and allowed for this.

This probably goes into the same direction as the previous discussion. Why is that the case? The benefit of the state model is that we have clearly defined transitions. It should be clear *who* invokes that transition and *why* / what the semantics are. Unless we have a really good reason why we would want to clear the output twice, it sounds as if we should keep the state diagram as easy as possible and drop this NOP transition.

Reply:
I fixed it as follows:

```cpp
void AbstractOperator::deregister_consumer() {
  DebugAssert(_consumer_count > 0, "Number of tracked consumer operators seems to be invalid.");
  // The following section is locked to prevent clear_output() from being called twice. Otherwise, a race condition
  // as follows might occur:
  //  1) T1 decreases _consumer_count, making it equal to one. After this operation, T1 gets suspended.
  //  2) T2 decreases _consumer_count as well, making it equal to zero. It enters the if statement and calls
  //     clear_output() for the first time.
  //  3) T1 wakes up and continues with the if statement. Since _consumer_count equals zero, it also calls
  //     clear_output(), which leads to an illegal state transition ExecutedAndCleared -> ExecutedAndCleared.
  std::lock_guard<std::mutex> lock(_deregister_consumer_mutex);

  _consumer_count--;
  if (_consumer_count == 0) clear_output();
}
```

However, locking deregister_consumer costs performance. For example:

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+--------------------------------------------------------------------------------------------+-------------$----------------------------------+
 | Parameter                     | benchmark_all_results/hyriseBenchmarkTPCH_fd52afb2fff07d264326398bac7f39ccaad522e5_mt.json | tpch_multi_s$10_50c.json                       |
 +-------------------------------+--------------------------------------------------------------------------------------------+-------------$----------------------------------+
 |  GIT-HASH                     | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                             | eb4c0450ef96$e98012e09345d90c205af5c69b8-dirty |
 |  benchmark_mode               | Ordered                                                                                    | Ordered
                                  |
 |  build_type                   | release                                                                                    | release
                                  |
 |  chunk_size                   | 65535                                                                                      | 65535
                                  |
 |  clients                      | 50                                                                                         | 50
                                  |
 |  clustering                   | None                                                                                       | None
                                  |
 |  compiler                     | gcc 9.2                                                                                    | gcc 9.2
                                  |
 |  cores                        | 0                                                                                          | 0
                                  |
 |  date                         | 2021-05-11 19:27:06                                                                        | 2021-05-13 2$:18:23                            |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                    | {'default': $'encoding': 'Dictionary'}}        |
 |  indexes                      | False                                                                                      | False
                                  |
 |  max_duration                 | 60000000000                                                                                | 60000000000
                                  |
 |  max_runs                     | -1                                                                                         | -1
                                  |
 |  scale_factor                 | 10.0                                                                                       | 10.0
                                  |
 |  time_unit                    | ns                                                                                         | ns
                                  |
 |  use_prepared_statements      | False                                                                                      | False
                                  |
 |  using_scheduler              | True                                                                                       | True
                                  |
 |  utilized_cores_per_numa_node | [20, 20, 20, 20]                                                                           | [20, 20, 20,
20]                               |
 |  verify                       | False                                                                                      | False
                                  |
 |  warmup_duration              | 0                                                                                          | 0
                                  |
 +-------------------------------+--------------------------------------------------------------------------------------------+------------------------------------------------+
```
</details>

```diff
 +----------++----------+----------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)   | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |      new |        ||      old |      new |        |         |
 +----------++----------+----------+--------++----------+----------+--------+---------+
-| TPC-H 01 ||  26926.7 |  28431.5 |   +6%  ||     1.47 |     1.40 |   -5%  |  0.3722 |
 | TPC-H 02 ||    124.4 |    123.9 |   -0%  ||   385.01 |   386.85 |   +0%  |  0.5453 |
 | TPC-H 03 ||   8996.7 |   8126.3 |  -10%  ||     4.68 |     4.70 |   +0%  |  0.1038 |
 | TPC-H 04 ||   6089.0 |   6228.9 |   +2%  ||     6.83 |     6.53 |   -4%  |  0.7088 |
-| TPC-H 05 ||  10584.0 |  11176.7 |   +6%  ||     3.83 |     3.53 |   -8%  |  0.3983 |
 | TPC-H 06 ||   1250.2 |   1219.3 |   -2%  ||    38.36 |    38.66 |   +1%  |  0.5326 |
 | TPC-H 07 ||   3264.1 |   3263.4 |   -0%  ||    14.32 |    14.26 |   -0%  |  0.9976 |
 | TPC-H 08 ||   2553.8 |   2424.7 |   -5%  ||    18.43 |    18.58 |   +1%  |  0.2712 |
-| TPC-H 09 ||  18409.3 |  17087.1 |   -7%  ||     1.92 |     1.82 |   -5%  |  0.1699 |
 | TPC-H 10 ||  10311.5 |  10910.2 |   +6%  ||     3.83 |     3.85 |   +0%  |  0.3253 |
 | TPC-H 11 ||    528.7 |    524.2 |   -1%  ||    92.86 |    93.71 |   +1%  |  0.4770 |
 | TPC-H 12 ||   3668.9 |   3659.8 |   -0%  ||    12.92 |    13.05 |   +1%  |  0.9595 |
 | TPC-H 13 ||  19150.3 |  19654.6 |   +3%  ||     1.93 |     1.85 |   -4%  |  0.6000 |
 | TPC-H 14 ||   1571.0 |   1616.3 |   +3%  ||    30.38 |    30.00 |   -1%  |  0.5106 |
 | TPC-H 15 ||   1566.7 |   1562.3 |   -0%  ||    31.32 |    31.13 |   -1%  |  0.9034 |
 | TPC-H 16 ||   2642.3 |   2634.7 |   -0%  ||    18.38 |    18.30 |   -0%  |  0.9003 |
 | TPC-H 17 ||   1409.5 |   1358.6 |   -4%  ||    34.48 |    35.20 |   +2%  |  0.4483 |
 | TPC-H 18 ||  24829.8 |  23448.4 |   -6%  ||     1.50 |     1.47 |   -2%  |  0.2824 |
 | TPC-H 19 ||   1391.0 |   1443.3 |   +4%  ||    33.68 |    33.60 |   -0%  |  0.3801 |
 | TPC-H 20 ||   1974.9 |   1980.3 |   +0%  ||    23.71 |    23.91 |   +1%  |  0.9493 |
 | TPC-H 21 ||  13125.3 |  13291.2 |   +1%  ||     2.73 |     2.75 |   +1%  |  0.8508 |
 | TPC-H 22 ||   1463.5 |   1460.0 |   -0%  ||    32.66 |    32.47 |   -1%  |  0.9540 |
 +----------++----------+----------+--------++----------+----------+--------+---------+
 | Sum      || 161831.8 | 161625.7 |   -0%  ||          |          |        |         |
 | Geomean  ||          |          |        ||          |          |   -1%  |         |
 +----------++----------+----------+--------++----------+----------+--------+---------+
```

I should re-run benchmark_all.sh due to this change.

Reply:
> I should re-run benchmark_all.sh due to this change.

Done. PR has been updated. 

Here are the results before locking deregister_consumer:

**System**
<details>
<summary>pella - click to expand</summary>

| property | value |
| -- | -- |
| Hostname | pella |
| CPU | Intel(R) Xeon(R) CPU E7- 8870  @ 2.40GHz |
| Memory | 1.3TB |
| numactl | nodebind: 0 1 2 3  |
| numactl | membind: 0 1 2 3  |
</details>

**Commit Info and Build Time**
| commit | date | message | build time |
| -- | -- | -- | -- |
| e1389f276 | 26.04.2021 19:18 | Markus' retirement (#2359) | real 772.32 user 3791.82 sys 183.49|
| fd52afb2f | 10.05.2021 18:44 | Fix CMakeLists.txt – add_subdirectory(plugins) | real 788.09 user 4025.27 sys 211.23|


**hyriseBenchmarkTPCH - single-threaded, SF 10.0**
<details>
<summary>
Sum of avg. item runtimes: -11%
 || 
Geometric mean of throughput changes: +13%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_e1389f276e4e0fbfec34eca098365999553c05f8_st.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_fd52afb2fff07d264326398bac7f39ccaad522e5_st.json |
 +--------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                             | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                             |
 |  benchmark_mode          | Ordered                                                                                                                                    | Ordered                                                                                                                                    |
 |  build_type              | release                                                                                                                                    | release                                                                                                                                    |
 |  chunk_size              | 65535                                                                                                                                      | 65535                                                                                                                                      |
 |  clients                 | 1                                                                                                                                          | 1                                                                                                                                          |
 |  clustering              | None                                                                                                                                       | None                                                                                                                                       |
 |  compiler                | gcc 9.2                                                                                                                                    | gcc 9.2                                                                                                                                    |
 |  cores                   | 0                                                                                                                                          | 0                                                                                                                                          |
 |  date                    | 2021-05-11 09:47:25                                                                                                                        | 2021-05-11 17:38:33                                                                                                                        |
 |  encoding                | {'default': {'encoding': 'Dictionary'}}                                                                                                    | {'default': {'encoding': 'Dictionary'}}                                                                                                    |
 |  indexes                 | False                                                                                                                                      | False                                                                                                                                      |
 |  max_duration            | 60000000000                                                                                                                                | 60000000000                                                                                                                                |
 |  max_runs                | -1                                                                                                                                         | -1                                                                                                                                         |
 |  scale_factor            | 10.0                                                                                                                                       | 10.0                                                                                                                                       |
 |  time_unit               | ns                                                                                                                                         | ns                                                                                                                                         |
 |  use_prepared_statements | False                                                                                                                                      | False                                                                                                                                      |
 |  using_scheduler         | False                                                                                                                                      | False                                                                                                                                      |
 |  verify                  | False                                                                                                                                      | False                                                                                                                                      |
 |  warmup_duration         | 0                                                                                                                                          | 0                                                                                                                                          |
 +--------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||  11362.4 | 11181.3 |   -2%  ||     0.09 |     0.09 |   +2%  |       ˅ |
+| TPC-H 02 ||     93.0 |    84.7 |   -9%  ||    10.75 |    11.81 |  +10%  |  0.0000 |
+| TPC-H 03 ||   4619.1 |  4271.8 |   -8%  ||     0.22 |     0.23 |   +8%  |  0.0000 |
+| TPC-H 04 ||   3385.4 |  3089.2 |   -9%  ||     0.30 |     0.32 |  +10%  |  0.0000 |
+| TPC-H 05 ||   7829.1 |  6852.0 |  -12%  ||     0.13 |     0.15 |  +14%  |       ˅ |
+| TPC-H 06 ||    437.0 |   389.4 |  -11%  ||     2.29 |     2.57 |  +12%  |  0.0000 |
+| TPC-H 07 ||   2110.7 |  1886.4 |  -11%  ||     0.47 |     0.53 |  +12%  |  0.0000 |
+| TPC-H 08 ||   1743.7 |  1542.3 |  -12%  ||     0.57 |     0.65 |  +13%  |  0.0000 |
+| TPC-H 09 ||  13882.1 | 11815.0 |  -15%  ||     0.07 |     0.08 |  +17%  |       ˅ |
+| TPC-H 10 ||   5378.5 |  4886.6 |   -9%  ||     0.19 |     0.20 |  +10%  |  0.0000 |
+| TPC-H 11 ||    230.1 |   200.7 |  -13%  ||     4.35 |     4.98 |  +15%  |  0.0000 |
+| TPC-H 12 ||   2477.7 |  2090.9 |  -16%  ||     0.40 |     0.48 |  +19%  |  0.0000 |
+| TPC-H 13 ||  12490.5 | 11035.5 |  -12%  ||     0.08 |     0.09 |  +13%  |       ˅ |
+| TPC-H 14 ||   1086.1 |   957.8 |  -12%  ||     0.92 |     1.04 |  +13%  |  0.0000 |
+| TPC-H 15 ||    618.0 |   507.4 |  -18%  ||     1.62 |     1.97 |  +22%  |  0.0000 |
+| TPC-H 16 ||   1403.5 |  1332.9 |   -5%  ||     0.71 |     0.75 |   +5%  |  0.0000 |
+| TPC-H 17 ||    675.9 |   571.1 |  -16%  ||     1.48 |     1.75 |  +18%  |  0.0000 |
 | TPC-H 18 ||   5300.0 |  5173.0 |   -2%  ||     0.19 |     0.19 |   +2%  |  0.0011 |
+| TPC-H 19 ||    776.2 |   673.5 |  -13%  ||     1.29 |     1.48 |  +15%  |  0.0000 |
+| TPC-H 20 ||   1102.4 |   913.3 |  -17%  ||     0.91 |     1.09 |  +21%  |  0.0000 |
+| TPC-H 21 ||  12342.2 | 10420.9 |  -16%  ||     0.08 |     0.10 |  +18%  |       ˅ |
+| TPC-H 22 ||   1052.5 |   944.1 |  -10%  ||     0.95 |     1.06 |  +11%  |  0.0000 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
+| Sum      ||  90396.2 | 80820.0 |  -11%  ||          |          |        |         |
+| Geomean  ||          |         |        ||          |          |  +13%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 |          || ˅ Insufficient number of runs for p-value calculation                 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCH - single-threaded, SF 0.01**
<details>
<summary>
Sum of avg. item runtimes: +2%
 || 
Geometric mean of throughput changes: -2%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----+------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_e1389f276e4e0fbfec34eca098365999553c05f8_st_s01.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_fd52afb2fff07d264326398bac7f39ccaad522e5_st_s01.json |
 +--------------------------+------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                                 | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                                 |
 |  benchmark_mode          | Ordered                                                                                                                                        | Ordered                                                                                                                                        |
 |  build_type              | release                                                                                                                                        | release                                                                                                                                        |
 |  chunk_size              | 65535                                                                                                                                          | 65535                                                                                                                                          |
 |  clients                 | 1                                                                                                                                              | 1                                                                                                                                              |
 |  clustering              | None                                                                                                                                           | None                                                                                                                                           |
 |  compiler                | gcc 9.2                                                                                                                                        | gcc 9.2                                                                                                                                        |
 |  cores                   | 0                                                                                                                                              | 0                                                                                                                                              |
 |  date                    | 2021-05-11 10:13:10                                                                                                                            | 2021-05-11 18:04:11                                                                                                                            |
 |  encoding                | {'default': {'encoding': 'Dictionary'}}                                                                                                        | {'default': {'encoding': 'Dictionary'}}                                                                                                        |
 |  indexes                 | False                                                                                                                                          | False                                                                                                                                          |
 |  max_duration            | 60000000000                                                                                                                                    | 60000000000                                                                                                                                    |
 |  max_runs                | -1                                                                                                                                             | -1                                                                                                                                             |
 |  scale_factor            | 0.009999999776482582                                                                                                                           | 0.009999999776482582                                                                                                                           |
 |  time_unit               | ns                                                                                                                                             | ns                                                                                                                                             |
 |  use_prepared_statements | False                                                                                                                                          | False                                                                                                                                          |
 |  using_scheduler         | False                                                                                                                                          | False                                                                                                                                          |
 |  verify                  | False                                                                                                                                          | False                                                                                                                                          |
 |  warmup_duration         | 0                                                                                                                                              | 0                                                                                                                                              |
 +--------------------------+------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||      9.5 |     9.3 |   -2%  ||   105.34 |   107.81 |   +2%  |  0.0000 |
 | TPC-H 02 ||      1.7 |     1.8 |   +4%  ||   584.04 |   563.06 |   -4%  |  0.0000 |
 | TPC-H 03 ||      1.4 |     1.4 |   +1%  ||   736.45 |   726.00 |   -1%  |  0.0000 |
 | TPC-H 04 ||      0.9 |     0.9 |   -0%  ||  1094.07 |  1094.58 |   +0%  |  0.0570 |
 | TPC-H 05 ||      2.1 |     2.2 |   +2%  ||   468.51 |   461.09 |   -2%  |  0.0000 |
 | TPC-H 06 ||      0.4 |     0.5 |   +2%  ||  2225.94 |  2182.34 |   -2%  |  0.0000 |
-| TPC-H 07 ||      3.1 |     3.3 |   +6%  ||   320.76 |   301.62 |   -6%  |  0.0004 |
 | TPC-H 08 ||     25.3 |    26.2 |   +4%  ||    39.48 |    38.13 |   -3%  |  0.0005 |
 | TPC-H 09 ||      4.0 |     4.0 |   +0%  ||   247.70 |   246.67 |   -0%  |  0.3628 |
 | TPC-H 10 ||      1.6 |     1.6 |   +2%  ||   635.52 |   621.85 |   -2%  |  0.0000 |
 | TPC-H 11 ||      0.6 |     0.6 |   +3%  ||  1717.74 |  1673.31 |   -3%  |  0.0000 |
 | TPC-H 12 ||      1.3 |     1.3 |   +2%  ||   767.32 |   753.56 |   -2%  |  0.0000 |
 | TPC-H 13 ||      3.5 |     3.5 |   +1%  ||   285.11 |   283.29 |   -1%  |  0.0000 |
 | TPC-H 14 ||      0.7 |     0.7 |   +2%  ||  1418.48 |  1388.81 |   -2%  |  0.0000 |
 | TPC-H 15 ||      2.3 |     2.4 |   +3%  ||   425.32 |   414.00 |   -3%  |  0.0000 |
 | TPC-H 16 ||      3.8 |     3.8 |   +1%  ||   265.81 |   263.06 |   -1%  |  0.0000 |
 | TPC-H 17 ||      0.7 |     0.7 |   +5%  ||  1432.48 |  1369.27 |   -4%  |  0.0000 |
 | TPC-H 18 ||      2.9 |     3.0 |   +2%  ||   340.87 |   334.04 |   -2%  |  0.0000 |
 | TPC-H 19 ||     10.1 |    10.2 |   +1%  ||    99.35 |    97.96 |   -1%  |  0.0000 |
 | TPC-H 20 ||      4.7 |     4.8 |   +3%  ||   213.51 |   207.72 |   -3%  |  0.0000 |
 | TPC-H 21 ||      7.3 |     7.3 |   +0%  ||   137.01 |   136.46 |   -0%  |  0.0312 |
 | TPC-H 22 ||      2.2 |     2.2 |   +3%  ||   459.41 |   447.49 |   -3%  |  0.0000 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||     90.1 |    91.8 |   +2%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   -2%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCH - single-threaded, SF 1.0**
<details>
<summary>
Sum of avg. item runtimes: +1%
 || 
Geometric mean of throughput changes: -2%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----+-----------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_e1389f276e4e0fbfec34eca098365999553c05f8_st_s1.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_fd52afb2fff07d264326398bac7f39ccaad522e5_st_s1.json |
 +--------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                                | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                                |
 |  benchmark_mode          | Ordered                                                                                                                                       | Ordered                                                                                                                                       |
 |  build_type              | release                                                                                                                                       | release                                                                                                                                       |
 |  chunk_size              | 65535                                                                                                                                         | 65535                                                                                                                                         |
 |  clients                 | 1                                                                                                                                             | 1                                                                                                                                             |
 |  clustering              | None                                                                                                                                          | None                                                                                                                                          |
 |  compiler                | gcc 9.2                                                                                                                                       | gcc 9.2                                                                                                                                       |
 |  cores                   | 0                                                                                                                                             | 0                                                                                                                                             |
 |  date                    | 2021-05-11 10:35:14                                                                                                                           | 2021-05-11 18:26:14                                                                                                                           |
 |  encoding                | {'default': {'encoding': 'Dictionary'}}                                                                                                       | {'default': {'encoding': 'Dictionary'}}                                                                                                       |
 |  indexes                 | False                                                                                                                                         | False                                                                                                                                         |
 |  max_duration            | 60000000000                                                                                                                                   | 60000000000                                                                                                                                   |
 |  max_runs                | -1                                                                                                                                            | -1                                                                                                                                            |
 |  scale_factor            | 1.0                                                                                                                                           | 1.0                                                                                                                                           |
 |  time_unit               | ns                                                                                                                                            | ns                                                                                                                                            |
 |  use_prepared_statements | False                                                                                                                                         | False                                                                                                                                         |
 |  using_scheduler         | False                                                                                                                                         | False                                                                                                                                         |
 |  verify                  | False                                                                                                                                         | False                                                                                                                                         |
 |  warmup_duration         | 0                                                                                                                                             | 0                                                                                                                                             |
 +--------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||   1053.7 |  1065.7 |   +1%  ||     0.95 |     0.94 |   -1%  |  0.3402 |
 | TPC-H 02 ||      9.4 |     9.4 |   +1%  ||   106.70 |   105.83 |   -1%  |  0.5107 |
 | TPC-H 03 ||    181.3 |   181.2 |   -0%  ||     5.52 |     5.52 |   +0%  |  0.8587 |
 | TPC-H 04 ||    110.2 |   110.4 |   +0%  ||     9.07 |     9.06 |   -0%  |  0.0672 |
 | TPC-H 05 ||    365.7 |   367.2 |   +0%  ||     2.73 |     2.72 |   -0%  |  0.3518 |
 | TPC-H 06 ||     34.4 |    34.5 |   +0%  ||    29.03 |    28.99 |   -0%  |  0.4518 |
 | TPC-H 07 ||    147.0 |   150.0 |   +2%  ||     6.80 |     6.67 |   -2%  |  0.0317 |
 | TPC-H 08 ||    122.3 |   124.0 |   +1%  ||     8.18 |     8.07 |   -1%  |  0.0213 |
 | TPC-H 09 ||    800.5 |   799.0 |   -0%  ||     1.25 |     1.25 |   +0%  |  0.6676 |
 | TPC-H 10 ||    222.7 |   223.0 |   +0%  ||     4.49 |     4.48 |   -0%  |  0.6029 |
 | TPC-H 11 ||     10.7 |    10.7 |   +0%  ||    93.19 |    93.03 |   -0%  |  0.2468 |
 | TPC-H 12 ||    136.4 |   137.0 |   +0%  ||     7.33 |     7.30 |   -0%  |  0.0040 |
 | TPC-H 13 ||    662.3 |   655.8 |   -1%  ||     1.51 |     1.52 |   +1%  |  0.0000 |
 | TPC-H 14 ||     70.5 |    70.1 |   -0%  ||    14.19 |    14.26 |   +0%  |  0.0006 |
 | TPC-H 15 ||     42.5 |    42.6 |   +0%  ||    23.52 |    23.48 |   -0%  |  0.0094 |
 | TPC-H 16 ||    124.7 |   127.0 |   +2%  ||     8.02 |     7.87 |   -2%  |  0.0000 |
-| TPC-H 17 ||     43.5 |    51.7 |  +19%  ||    22.97 |    19.34 |  -16%  |  0.0000 |
 | TPC-H 18 ||    523.2 |   526.5 |   +1%  ||     1.91 |     1.90 |   -1%  |  0.1761 |
-| TPC-H 19 ||     66.0 |    74.9 |  +14%  ||    15.16 |    13.35 |  -12%  |  0.0000 |
-| TPC-H 20 ||     78.0 |    92.0 |  +18%  ||    12.81 |    10.86 |  -15%  |  0.0000 |
 | TPC-H 21 ||    616.8 |   604.5 |   -2%  ||     1.62 |     1.65 |   +2%  |  0.3452 |
 | TPC-H 22 ||     69.2 |    69.1 |   -0%  ||    14.44 |    14.48 |   +0%  |  0.0412 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||   5491.0 |  5526.3 |   +1%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   -2%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCH - multi-threaded, clustering config 'Pruning', SF 10.0**
<details>
<summary>
Sum of avg. item runtimes: +1%
 || 
Geometric mean of throughput changes: -0%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                     | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_e1389f276e4e0fbfec34eca098365999553c05f8_mt_clustered.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_fd52afb2fff07d264326398bac7f39ccaad522e5_mt_clustered.json |
 +-------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                     | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                                       | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                                       |
 |  benchmark_mode               | Ordered                                                                                                                                              | Ordered                                                                                                                                              |
 |  build_type                   | release                                                                                                                                              | release                                                                                                                                              |
 |  chunk_size                   | 65535                                                                                                                                                | 65535                                                                                                                                                |
 |  clients                      | 50                                                                                                                                                   | 50                                                                                                                                                   |
 |  clustering                   | Pruning                                                                                                                                              | Pruning                                                                                                                                              |
 |  compiler                     | gcc 9.2                                                                                                                                              | gcc 9.2                                                                                                                                              |
 |  cores                        | 0                                                                                                                                                    | 0                                                                                                                                                    |
 |  date                         | 2021-05-11 10:57:34                                                                                                                                  | 2021-05-11 18:48:35                                                                                                                                  |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                                                                              | {'default': {'encoding': 'Dictionary'}}                                                                                                              |
 |  indexes                      | False                                                                                                                                                | False                                                                                                                                                |
 |  max_duration                 | 60000000000                                                                                                                                          | 60000000000                                                                                                                                          |
 |  max_runs                     | -1                                                                                                                                                   | -1                                                                                                                                                   |
 |  scale_factor                 | 10.0                                                                                                                                                 | 10.0                                                                                                                                                 |
 |  time_unit                    | ns                                                                                                                                                   | ns                                                                                                                                                   |
 |  use_prepared_statements      | False                                                                                                                                                | False                                                                                                                                                |
 |  using_scheduler              | True                                                                                                                                                 | True                                                                                                                                                 |
 |  utilized_cores_per_numa_node | [20, 20, 20, 20]                                                                                                                                     | [20, 20, 20, 20]                                                                                                                                     |
 |  verify                       | False                                                                                                                                                | False                                                                                                                                                |
 |  warmup_duration              | 0                                                                                                                                                    | 0                                                                                                                                                    |
 +-------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+----------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)   | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |      new |        ||      old |      new |        |         |
 +----------++----------+----------+--------++----------+----------+--------+---------+
-| TPC-H 01 ||  24489.3 |  25235.7 |   +3%  ||     1.67 |     1.55 |   -7%  |  0.6001 |
 | TPC-H 02 ||    122.5 |    124.9 |   +2%  ||   390.64 |   383.38 |   -2%  |  0.0157 |
 | TPC-H 03 ||   5465.1 |   5239.9 |   -4%  ||     8.20 |     8.28 |   +1%  |  0.4500 |
 | TPC-H 04 ||   7661.7 |   7342.8 |   -4%  ||     5.73 |     5.98 |   +4%  |  0.5055 |
+| TPC-H 05 ||  12486.0 |  12066.0 |   -3%  ||     3.15 |     3.30 |   +5%  |  0.6207 |
 | TPC-H 06 ||    110.5 |    110.9 |   +0%  ||   429.54 |   428.73 |   -0%  |  0.6973 |
 | TPC-H 07 ||   1915.4 |   1876.5 |   -2%  ||    25.57 |    25.52 |   -0%  |  0.5732 |
 | TPC-H 08 ||   1576.2 |   1591.7 |   +1%  ||    30.02 |    29.45 |   -2%  |  0.8268 |
-| TPC-H 09 ||  19237.0 |  20354.9 |   +6%  ||     1.78 |     1.68 |   -6%  |  0.3438 |
 | TPC-H 10 ||   8002.4 |   8000.6 |   -0%  ||     5.58 |     5.37 |   -4%  |  0.9961 |
 | TPC-H 11 ||    392.8 |    388.5 |   -1%  ||   125.09 |   126.41 |   +1%  |  0.3066 |
 | TPC-H 12 ||   1790.8 |   1912.2 |   +7%  ||    25.76 |    25.50 |   -1%  |  0.0897 |
+| TPC-H 13 ||  19976.1 |  19314.0 |   -3%  ||     1.73 |     1.87 |   +8%  |  0.4616 |
 | TPC-H 14 ||    801.6 |    803.5 |   +0%  ||    61.11 |    60.71 |   -1%  |  0.9272 |
 | TPC-H 15 ||    335.7 |    328.4 |   -2%  ||   146.21 |   149.58 |   +2%  |  0.0112 |
 | TPC-H 16 ||   2606.9 |   2612.5 |   +0%  ||    18.63 |    18.45 |   -1%  |  0.9258 |
 | TPC-H 17 ||    801.9 |    804.7 |   +0%  ||    60.70 |    60.29 |   -1%  |  0.9087 |
-| TPC-H 18 ||  38752.5 |  40163.0 |   +4%  ||     0.82 |     0.75 |   -8%  |  0.4745 |
 | TPC-H 19 ||    903.7 |    899.8 |   -0%  ||    53.71 |    53.69 |   -0%  |  0.8878 |
 | TPC-H 20 ||   1028.7 |    976.1 |   -5%  ||    47.36 |    48.64 |   +3%  |  0.0620 |
+| TPC-H 21 ||  16059.8 |  15490.4 |   -4%  ||     2.18 |     2.33 |   +7%  |  0.6114 |
 | TPC-H 22 ||   1304.8 |   1306.3 |   +0%  ||    37.08 |    37.14 |   +0%  |  0.9759 |
 +----------++----------+----------+--------++----------+----------+--------+---------+
 | Sum      || 165821.5 | 166943.1 |   +1%  ||          |          |        |         |
 | Geomean  ||          |          |        ||          |          |   -0%  |         |
 +----------++----------+----------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCH - multi-threaded (50 clients), SF 10.0**
<details>
<summary>
Sum of avg. item runtimes: -2%
 || 
Geometric mean of throughput changes: +2%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                     | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_e1389f276e4e0fbfec34eca098365999553c05f8_mt.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCH_fd52afb2fff07d264326398bac7f39ccaad522e5_mt.json |
 +-------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                     | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                             | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                             |
 |  benchmark_mode               | Ordered                                                                                                                                    | Ordered                                                                                                                                    |
 |  build_type                   | release                                                                                                                                    | release                                                                                                                                    |
 |  chunk_size                   | 65535                                                                                                                                      | 65535                                                                                                                                      |
 |  clients                      | 50                                                                                                                                         | 50                                                                                                                                         |
 |  clustering                   | None                                                                                                                                       | None                                                                                                                                       |
 |  compiler                     | gcc 9.2                                                                                                                                    | gcc 9.2                                                                                                                                    |
 |  cores                        | 0                                                                                                                                          | 0                                                                                                                                          |
 |  date                         | 2021-05-11 11:34:02                                                                                                                        | 2021-05-11 19:27:06                                                                                                                        |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                                                                    | {'default': {'encoding': 'Dictionary'}}                                                                                                    |
 |  indexes                      | False                                                                                                                                      | False                                                                                                                                      |
 |  max_duration                 | 60000000000                                                                                                                                | 60000000000                                                                                                                                |
 |  max_runs                     | -1                                                                                                                                         | -1                                                                                                                                         |
 |  scale_factor                 | 10.0                                                                                                                                       | 10.0                                                                                                                                       |
 |  time_unit                    | ns                                                                                                                                         | ns                                                                                                                                         |
 |  use_prepared_statements      | False                                                                                                                                      | False                                                                                                                                      |
 |  using_scheduler              | True                                                                                                                                       | True                                                                                                                                       |
 |  utilized_cores_per_numa_node | [20, 20, 20, 20]                                                                                                                           | [20, 20, 20, 20]                                                                                                                           |
 |  verify                       | False                                                                                                                                      | False                                                                                                                                      |
 |  warmup_duration              | 0                                                                                                                                          | 0                                                                                                                                          |
 +-------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+----------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)   | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |      new |        ||      old |      new |        |         |
 +----------++----------+----------+--------++----------+----------+--------+---------+
+| TPC-H 01 ||  27806.6 |  26926.7 |   -3%  ||     1.40 |     1.47 |   +5%  |  0.5836 |
 | TPC-H 02 ||    122.8 |    124.4 |   +1%  ||   390.10 |   385.01 |   -1%  |  0.0585 |
+| TPC-H 03 ||   7723.8 |   8996.7 |  +16%  ||     4.38 |     4.68 |   +7%  |  0.0255 |
+| TPC-H 04 ||   6699.3 |   6089.0 |   -9%  ||     6.47 |     6.83 |   +6%  |  0.1052 |
 | TPC-H 05 ||  10488.1 |  10584.0 |   +1%  ||     3.77 |     3.83 |   +2%  |  0.8884 |
 | TPC-H 06 ||   1274.0 |   1250.2 |   -2%  ||    38.29 |    38.36 |   +0%  |  0.6180 |
 | TPC-H 07 ||   3177.0 |   3264.1 |   +3%  ||    13.93 |    14.32 |   +3%  |  0.6831 |
 | TPC-H 08 ||   2554.3 |   2553.8 |   -0%  ||    18.22 |    18.43 |   +1%  |  0.9971 |
 | TPC-H 09 ||  18016.0 |  18409.3 |   +2%  ||     1.88 |     1.92 |   +2%  |  0.6946 |
 | TPC-H 10 ||  10844.9 |  10311.5 |   -5%  ||     3.90 |     3.83 |   -2%  |  0.3853 |
 | TPC-H 11 ||    530.5 |    528.7 |   -0%  ||    92.75 |    92.86 |   +0%  |  0.7850 |
 | TPC-H 12 ||   3545.5 |   3668.9 |   +3%  ||    12.45 |    12.92 |   +4%  |  0.4774 |
+| TPC-H 13 ||  21593.1 |  19150.3 |  -11%  ||     1.67 |     1.93 |  +16%  |  0.0317 |
 | TPC-H 14 ||   1579.3 |   1571.0 |   -1%  ||    30.46 |    30.38 |   -0%  |  0.8875 |
 | TPC-H 15 ||   1574.6 |   1566.7 |   -0%  ||    31.03 |    31.32 |   +1%  |  0.8302 |
 | TPC-H 16 ||   2649.3 |   2642.3 |   -0%  ||    18.42 |    18.38 |   -0%  |  0.9060 |
 | TPC-H 17 ||   1375.8 |   1409.5 |   +2%  ||    35.11 |    34.48 |   -2%  |  0.6052 |
+| TPC-H 18 ||  25896.6 |  24829.8 |   -4%  ||     1.43 |     1.50 |   +5%  |  0.4054 |
 | TPC-H 19 ||   1450.9 |   1391.0 |   -4%  ||    33.50 |    33.68 |   +1%  |  0.2811 |
 | TPC-H 20 ||   1980.1 |   1974.9 |   -0%  ||    24.25 |    23.71 |   -2%  |  0.9497 |
 | TPC-H 21 ||  12814.5 |  13125.3 |   +2%  ||     2.70 |     2.73 |   +1%  |  0.7306 |
 | TPC-H 22 ||   1483.9 |   1463.5 |   -1%  ||    32.70 |    32.66 |   -0%  |  0.7228 |
 +----------++----------+----------+--------++----------+----------+--------+---------+
 | Sum      || 165180.9 | 161831.8 |   -2%  ||          |          |        |         |
 | Geomean  ||          |          |        ||          |          |   +2%  |         |
 +----------++----------+----------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCDS - single-threaded**
<details>
<summary>
Sum of avg. item runtimes: -3%
 || 
Geometric mean of throughput changes: +2%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter        | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCDS_e1389f276e4e0fbfec34eca098365999553c05f8_st.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCDS_fd52afb2fff07d264326398bac7f39ccaad522e5_st.json |
 +------------------+---------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH        | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                              | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                              |
 |  benchmark_mode  | Ordered                                                                                                                                     | Ordered                                                                                                                                     |
 |  build_type      | release                                                                                                                                     | release                                                                                                                                     |
 |  chunk_size      | 65535                                                                                                                                       | 65535                                                                                                                                       |
 |  clients         | 1                                                                                                                                           | 1                                                                                                                                           |
 |  compiler        | gcc 9.2                                                                                                                                     | gcc 9.2                                                                                                                                     |
 |  cores           | 0                                                                                                                                           | 0                                                                                                                                           |
 |  date            | 2021-05-11 12:02:52                                                                                                                         | 2021-05-11 19:56:12                                                                                                                         |
 |  encoding        | {'default': {'encoding': 'Dictionary'}}                                                                                                     | {'default': {'encoding': 'Dictionary'}}                                                                                                     |
 |  indexes         | False                                                                                                                                       | False                                                                                                                                       |
 |  max_duration    | 60000000000                                                                                                                                 | 60000000000                                                                                                                                 |
 |  max_runs        | -1                                                                                                                                          | -1                                                                                                                                          |
 |  time_unit       | ns                                                                                                                                          | ns                                                                                                                                          |
 |  using_scheduler | False                                                                                                                                       | False                                                                                                                                       |
 |  verify          | False                                                                                                                                       | False                                                                                                                                       |
 |  warmup_duration | 0                                                                                                                                           | 0                                                                                                                                           |
 +------------------+---------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |     new |        ||      old |      new |        |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | 01      ||     44.5 |    42.7 |   -4%  ||    22.46 |    23.40 |   +4%  |  0.0000 |
 | 03      ||     10.8 |    10.8 |   +0%  ||    92.54 |    92.17 |   -0%  |  0.0000 |
 | 06      ||     30.5 |    30.1 |   -1%  ||    32.77 |    33.24 |   +1%  |  0.0000 |
 | 07      ||     60.2 |    58.7 |   -3%  ||    16.60 |    17.04 |   +3%  |  0.0000 |
 | 09      ||    158.3 |   158.1 |   -0%  ||     6.32 |     6.32 |   +0%  |  0.7126 |
 | 10      ||     36.1 |    36.1 |   +0%  ||    27.69 |    27.66 |   -0%  |  0.4787 |
 | 13      ||    105.2 |   104.9 |   -0%  ||     9.50 |     9.53 |   +0%  |  0.8420 |
 | 15      ||     17.9 |    17.7 |   -1%  ||    55.97 |    56.55 |   +1%  |  0.0000 |
 | 17      ||     53.2 |    53.0 |   -0%  ||    18.81 |    18.85 |   +0%  |  0.8255 |
 | 19      ||     15.2 |    15.1 |   -1%  ||    65.65 |    66.13 |   +1%  |  0.0000 |
 | 25      ||     25.9 |    25.7 |   -1%  ||    38.60 |    38.86 |   +1%  |  0.5353 |
 | 26      ||     27.2 |    27.1 |   -0%  ||    36.82 |    36.94 |   +0%  |  0.0219 |
 | 28      ||    602.5 |   600.2 |   -0%  ||     1.66 |     1.67 |   +0%  |  0.0113 |
 | 29      ||     45.5 |    45.2 |   -1%  ||    21.96 |    22.10 |   +1%  |  0.5460 |
 | 31      ||    221.0 |   220.6 |   -0%  ||     4.52 |     4.53 |   +0%  |  0.4118 |
 | 34      ||     29.2 |    29.1 |   -0%  ||    34.25 |    34.38 |   +0%  |  0.0002 |
+| 35      ||    134.3 |   124.1 |   -8%  ||     7.45 |     8.06 |   +8%  |  0.0000 |
+| 39a     ||    267.1 |   245.1 |   -8%  ||     3.74 |     4.08 |   +9%  |  0.0000 |
+| 39b     ||    261.7 |   245.4 |   -6%  ||     3.82 |     4.07 |   +7%  |  0.0000 |
 | 41      ||     59.8 |    60.4 |   +1%  ||    16.73 |    16.55 |   -1%  |  0.0000 |
 | 42      ||     12.0 |    12.0 |   -0%  ||    83.16 |    83.47 |   +0%  |  0.0000 |
 | 43      ||    417.0 |   406.5 |   -3%  ||     2.40 |     2.46 |   +3%  |  0.0000 |
 | 45      ||     16.4 |    16.3 |   -1%  ||    60.98 |    61.33 |   +1%  |  0.0000 |
+| 48      ||    123.3 |   114.5 |   -7%  ||     8.11 |     8.73 |   +8%  |  0.0000 |
 | 50      ||     17.0 |    16.9 |   -0%  ||    58.97 |    59.20 |   +0%  |  0.0029 |
 | 52      ||     12.2 |    12.2 |   -0%  ||    81.77 |    82.12 |   +0%  |  0.0000 |
 | 55      ||     11.9 |    11.9 |   -0%  ||    84.01 |    84.31 |   +0%  |  0.0000 |
 | 62      ||     96.7 |    96.9 |   +0%  ||    10.34 |    10.32 |   -0%  |  0.0305 |
+| 65      ||    122.3 |   114.5 |   -6%  ||     8.18 |     8.73 |   +7%  |  0.0000 |
+| 69      ||     33.2 |    30.9 |   -7%  ||    30.12 |    32.39 |   +8%  |  0.0000 |
 | 73      ||     15.6 |    15.6 |   +0%  ||    63.99 |    63.93 |   -0%  |  0.0737 |
+| 79      ||     84.1 |    75.7 |  -10%  ||    11.89 |    13.20 |  +11%  |  0.0000 |
 | 81      ||     27.7 |    27.6 |   -0%  ||    36.07 |    36.20 |   +0%  |  0.0000 |
 | 83      ||     14.8 |    14.9 |   +0%  ||    67.53 |    67.26 |   -0%  |  0.0000 |
 | 85      ||     64.5 |    62.7 |   -3%  ||    15.51 |    15.95 |   +3%  |  0.4008 |
 | 88      ||    104.4 |   104.7 |   +0%  ||     9.58 |     9.55 |   -0%  |  0.1934 |
 | 91      ||     18.9 |    19.5 |   +3%  ||    52.84 |    51.15 |   -3%  |  0.0000 |
+| 93      ||    494.4 |   468.2 |   -5%  ||     2.02 |     2.14 |   +6%  |  0.0000 |
 | 96      ||     11.1 |    11.1 |   -0%  ||    90.00 |    90.31 |   +0%  |  0.0000 |
 | 97      ||    627.9 |   627.0 |   -0%  ||     1.59 |     1.59 |   +0%  |  0.6007 |
 | 99      ||    201.6 |   199.4 |   -1%  ||     4.96 |     5.01 |   +1%  |  0.0000 |
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Sum     ||   4733.0 |  4609.3 |   -3%  ||          |          |        |         |
 | Geomean ||          |         |        ||          |          |   +2%  |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCDS - multi-threaded (50 clients)**
<details>
<summary>
Sum of avg. item runtimes: +1%
 || 
Geometric mean of throughput changes: -1%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+---------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                     | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCDS_e1389f276e4e0fbfec34eca098365999553c05f8_mt.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCDS_fd52afb2fff07d264326398bac7f39ccaad522e5_mt.json |
 +-------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                     | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                              | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                              |
 |  benchmark_mode               | Ordered                                                                                                                                     | Ordered                                                                                                                                     |
 |  build_type                   | release                                                                                                                                     | release                                                                                                                                     |
 |  chunk_size                   | 65535                                                                                                                                       | 65535                                                                                                                                       |
 |  clients                      | 50                                                                                                                                          | 50                                                                                                                                          |
 |  compiler                     | gcc 9.2                                                                                                                                     | gcc 9.2                                                                                                                                     |
 |  cores                        | 0                                                                                                                                           | 0                                                                                                                                           |
 |  date                         | 2021-05-11 12:43:58                                                                                                                         | 2021-05-11 20:37:18                                                                                                                         |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                                                                     | {'default': {'encoding': 'Dictionary'}}                                                                                                     |
 |  indexes                      | False                                                                                                                                       | False                                                                                                                                       |
 |  max_duration                 | 60000000000                                                                                                                                 | 60000000000                                                                                                                                 |
 |  max_runs                     | -1                                                                                                                                          | -1                                                                                                                                          |
 |  time_unit                    | ns                                                                                                                                          | ns                                                                                                                                          |
 |  using_scheduler              | True                                                                                                                                        | True                                                                                                                                        |
 |  utilized_cores_per_numa_node | [20, 20, 20, 20]                                                                                                                            | [20, 20, 20, 20]                                                                                                                            |
 |  verify                       | False                                                                                                                                       | False                                                                                                                                       |
 |  warmup_duration              | 0                                                                                                                                           | 0                                                                                                                                           |
 +-------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |     new |        ||      old |      new |        |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | 01      ||    110.1 |   108.6 |   -1%  ||   434.72 |   439.84 |   +1%  |  0.0024 |
 | 03      ||     24.5 |    24.6 |   +0%  ||  1733.08 |  1728.57 |   -0%  |  0.0636 |
 | 06      ||     64.5 |    64.0 |   -1%  ||   721.51 |   726.38 |   +1%  |  0.1292 |
 | 07      ||     99.4 |   100.1 |   +1%  ||   478.37 |   474.47 |   -1%  |  0.1900 |
+| 09      ||    354.3 |   330.9 |   -7%  ||   138.58 |   147.45 |   +6%  |  0.0000 |
 | 10      ||     53.2 |    53.8 |   +1%  ||   856.95 |   847.82 |   -1%  |  0.0323 |
 | 13      ||    261.7 |   260.8 |   -0%  ||   186.95 |   187.44 |   +0%  |  0.7101 |
 | 15      ||     46.3 |    45.2 |   -2%  ||   992.27 |  1015.28 |   +2%  |  0.0000 |
 | 17      ||    101.7 |   102.3 |   +1%  ||   467.37 |   465.40 |   -0%  |  0.4014 |
 | 19      ||     39.9 |    40.6 |   +2%  ||  1108.69 |  1093.80 |   -1%  |  0.0000 |
 | 25      ||     62.8 |    63.8 |   +1%  ||   736.97 |   727.42 |   -1%  |  0.0057 |
 | 26      ||     55.8 |    55.9 |   +0%  ||   827.22 |   824.67 |   -0%  |  0.3998 |
 | 28      ||   1401.4 |  1341.5 |   -4%  ||    33.85 |    33.88 |   +0%  |  0.3834 |
 | 29      ||     76.3 |    75.6 |   -1%  ||   616.43 |   622.74 |   +1%  |  0.0827 |
 | 31      ||    355.6 |   367.6 |   +3%  ||   137.78 |   133.37 |   -3%  |  0.1277 |
 | 34      ||     60.4 |    61.4 |   +2%  ||   765.38 |   754.03 |   -1%  |  0.0000 |
 | 35      ||    268.3 |   267.9 |   -0%  ||   182.47 |   182.34 |   -0%  |  0.8833 |
 | 39a     ||    484.2 |   493.2 |   +2%  ||   101.12 |    99.68 |   -1%  |  0.2278 |
 | 39b     ||    488.3 |   486.0 |   -0%  ||   100.53 |   101.29 |   +1%  |  0.7801 |
 | 41      ||    547.6 |   557.9 |   +2%  ||    89.87 |    87.69 |   -2%  |  0.3214 |
-| 42      ||     23.9 |    25.6 |   +7%  ||  1781.56 |  1680.69 |   -6%  |  0.0000 |
 | 43      ||    592.4 |   600.6 |   +1%  ||    83.13 |    81.88 |   -2%  |  0.3398 |
 | 45      ||     30.9 |    30.7 |   -1%  ||  1382.68 |  1385.90 |   +0%  |  0.0286 |
 | 48      ||    259.2 |   262.4 |   +1%  ||   188.31 |   186.36 |   -1%  |  0.3263 |
 | 50      ||     51.4 |    52.1 |   +1%  ||   887.23 |   875.20 |   -1%  |  0.0001 |
-| 52      ||     24.5 |    26.4 |   +8%  ||  1746.50 |  1641.81 |   -6%  |  0.0000 |
-| 55      ||     22.8 |    24.4 |   +7%  ||  1845.22 |  1741.09 |   -6%  |  0.0000 |
 | 62      ||    172.7 |   173.7 |   +1%  ||   280.89 |   279.05 |   -1%  |  0.3357 |
 | 65      ||    264.1 |   268.4 |   +2%  ||   185.27 |   182.39 |   -2%  |  0.0116 |
 | 69      ||     52.0 |    52.3 |   +1%  ||   876.98 |   873.53 |   -0%  |  0.3503 |
 | 73      ||     34.2 |    35.5 |   +4%  ||  1254.15 |  1213.80 |   -3%  |  0.0000 |
 | 79      ||    158.0 |   155.3 |   -2%  ||   306.25 |   311.40 |   +2%  |  0.0025 |
 | 81      ||     73.1 |    71.2 |   -2%  ||   640.98 |   656.21 |   +2%  |  0.0000 |
 | 83      ||     32.7 |    33.6 |   +3%  ||  1306.04 |  1270.50 |   -3%  |  0.0000 |
 | 85      ||    130.2 |   131.8 |   +1%  ||   369.01 |   364.77 |   -1%  |  0.2873 |
 | 88      ||    187.1 |   185.7 |   -1%  ||   258.36 |   257.99 |   -0%  |  0.7551 |
 | 91      ||     56.7 |    57.3 |   +1%  ||   811.53 |   803.38 |   -1%  |  0.0011 |
-| 93      ||    894.3 |   963.5 |   +8%  ||    54.79 |    51.04 |   -7%  |  0.0000 |
 | 96      ||     24.1 |    24.1 |   +0%  ||  1764.00 |  1764.55 |   +0%  |  0.7911 |
-| 97      ||   1372.7 |  1436.0 |   +5%  ||    35.66 |    33.85 |   -5%  |  0.0154 |
 | 99      ||    353.0 |   355.0 |   +1%  ||   139.07 |   138.33 |   -1%  |  0.4859 |
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Sum     ||   9766.3 |  9867.6 |   +1%  ||          |          |        |         |
 | Geomean ||          |         |        ||          |          |   -1%  |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCC - single-threaded**
<details>
<summary>
Sum of avg. item runtimes: +1%
 || 
Geometric mean of throughput changes: -1%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview-----------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter        | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCC_e1389f276e4e0fbfec34eca098365999553c05f8_st.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCC_fd52afb2fff07d264326398bac7f39ccaad522e5_st.json |
 +------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH        | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                             | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                             |
 |  benchmark_mode  | Shuffled                                                                                                                                   | Shuffled                                                                                                                                   |
 |  build_type      | release                                                                                                                                    | release                                                                                                                                    |
 |  chunk_size      | 65535                                                                                                                                      | 65535                                                                                                                                      |
 |  clients         | 1                                                                                                                                          | 1                                                                                                                                          |
 |  compiler        | gcc 9.2                                                                                                                                    | gcc 9.2                                                                                                                                    |
 |  cores           | 0                                                                                                                                          | 0                                                                                                                                          |
 |  date            | 2021-05-11 13:25:20                                                                                                                        | 2021-05-11 21:18:39                                                                                                                        |
 |  encoding        | {'default': {'encoding': 'Dictionary'}}                                                                                                    | {'default': {'encoding': 'Dictionary'}}                                                                                                    |
 |  indexes         | False                                                                                                                                      | False                                                                                                                                      |
 |  max_duration    | 60000000000                                                                                                                                | 60000000000                                                                                                                                |
 |  max_runs        | -1                                                                                                                                         | -1                                                                                                                                         |
 |  scale_factor    | 1                                                                                                                                          | 1                                                                                                                                          |
 |  time_unit       | ns                                                                                                                                         | ns                                                                                                                                         |
 |  using_scheduler | False                                                                                                                                      | False                                                                                                                                      |
 |  verify          | False                                                                                                                                      | False                                                                                                                                      |
 |  warmup_duration | 0                                                                                                                                          | 0                                                                                                                                          |
 +------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Item         || Latency (ms/iter)  | Change || Throughput (iter/s) | Change |              p-value |
 |              ||      old |     new |        ||      old |      new |        |                      |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Delivery     ||     65.6 |    66.7 |   +2%  ||     1.62 |     1.62 |   -0%  | (run time too short) |
 | New-Order    ||     42.3 |    42.3 |   +0%  ||    18.32 |    18.25 |   -0%  | (run time too short) |
 | Order-Status ||      2.9 |     2.9 |   +3%  ||     1.63 |     1.60 |   -2%  | (run time too short) |
 | Payment      ||      5.7 |     5.8 |   +2%  ||    17.62 |    17.33 |   -2%  | (run time too short) |
 | Stock-Level  ||      7.8 |     8.0 |   +3%  ||     1.65 |     1.60 |   -3%  | (run time too short) |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Sum          ||    124.2 |   125.8 |   +1%  ||          |          |        |                      |
 | Geomean      ||          |         |        ||          |          |   -1%  |                      |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
```
</details>


**hyriseBenchmarkTPCC - multi-threaded (50 clients)**
<details>
<summary>
Sum of avg. item runtimes: +1%
 || 
Geometric mean of throughput changes: -3%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                     | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCC_e1389f276e4e0fbfec34eca098365999553c05f8_mt.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkTPCC_fd52afb2fff07d264326398bac7f39ccaad522e5_mt.json |
 +-------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                     | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                             | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                             |
 |  benchmark_mode               | Shuffled                                                                                                                                   | Shuffled                                                                                                                                   |
 |  build_type                   | release                                                                                                                                    | release                                                                                                                                    |
 |  chunk_size                   | 65535                                                                                                                                      | 65535                                                                                                                                      |
 |  clients                      | 50                                                                                                                                         | 50                                                                                                                                         |
 |  compiler                     | gcc 9.2                                                                                                                                    | gcc 9.2                                                                                                                                    |
 |  cores                        | 0                                                                                                                                          | 0                                                                                                                                          |
 |  date                         | 2021-05-11 13:26:21                                                                                                                        | 2021-05-11 21:19:41                                                                                                                        |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                                                                    | {'default': {'encoding': 'Dictionary'}}                                                                                                    |
 |  indexes                      | False                                                                                                                                      | False                                                                                                                                      |
 |  max_duration                 | 60000000000                                                                                                                                | 60000000000                                                                                                                                |
 |  max_runs                     | -1                                                                                                                                         | -1                                                                                                                                         |
 |  scale_factor                 | 1                                                                                                                                          | 1                                                                                                                                          |
 |  time_unit                    | ns                                                                                                                                         | ns                                                                                                                                         |
 |  using_scheduler              | True                                                                                                                                       | True                                                                                                                                       |
 |  utilized_cores_per_numa_node | [20, 20, 20, 20]                                                                                                                           | [20, 20, 20, 20]                                                                                                                           |
 |  verify                       | False                                                                                                                                      | False                                                                                                                                      |
 |  warmup_duration              | 0                                                                                                                                          | 0                                                                                                                                          |
 +-------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Item         || Latency (ms/iter)  | Change || Throughput (iter/s) | Change |              p-value |
 |              ||      old |     new |        ||      old |      new |        |                      |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Delivery     ||    177.2 |   179.0 |   +1%  ||     5.44 |     5.39 |   -1%  | (run time too short) |
 |    unsucc.:  ||      3.4 |     3.7 |   +7%  ||   132.54 |   132.17 |   -0%  |                      |
 | New-Order    ||    100.9 |   101.9 |   +1%  ||    85.85 |    86.13 |   +0%  |               0.1478 |
 |    unsucc.:  ||      3.8 |     3.8 |   -0%  ||  1466.53 |  1461.45 |   -0%  |                      |
 | Order-Status ||      7.7 |     7.8 |   +1%  ||   138.02 |   137.59 |   -0%  |               0.0054 |
-| Payment      ||     14.8 |    15.1 |   +2%  ||     5.36 |     4.68 |  -13%  | (run time too short) |
 |    unsucc.:  ||      3.5 |     3.5 |   -0%  ||  1478.23 |  1474.26 |   -0%  |                      |
 | Stock-Level  ||     14.2 |    14.2 |   +0%  ||   137.97 |   137.54 |   -0%  |               0.7777 |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Sum          ||    314.8 |   318.1 |   +1%  ||          |          |        |                      |
 | Geomean      ||          |         |        ||          |          |   -3%  |                      |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
```
</details>


**hyriseBenchmarkJoinOrder - single-threaded**
<details>
<summary>
Sum of avg. item runtimes: -4%
 || 
Geometric mean of throughput changes: +6%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter        | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkJoinOrder_e1389f276e4e0fbfec34eca098365999553c05f8_st.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkJoinOrder_fd52afb2fff07d264326398bac7f39ccaad522e5_st.json |
 +------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH        | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                                  | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                                  |
 |  benchmark_mode  | Ordered                                                                                                                                         | Ordered                                                                                                                                         |
 |  build_type      | release                                                                                                                                         | release                                                                                                                                         |
 |  chunk_size      | 65535                                                                                                                                           | 65535                                                                                                                                           |
 |  clients         | 1                                                                                                                                               | 1                                                                                                                                               |
 |  compiler        | gcc 9.2                                                                                                                                         | gcc 9.2                                                                                                                                         |
 |  cores           | 0                                                                                                                                               | 0                                                                                                                                               |
 |  date            | 2021-05-11 13:27:24                                                                                                                             | 2021-05-11 21:20:44                                                                                                                             |
 |  encoding        | {'default': {'encoding': 'Dictionary'}}                                                                                                         | {'default': {'encoding': 'Dictionary'}}                                                                                                         |
 |  indexes         | False                                                                                                                                           | False                                                                                                                                           |
 |  max_duration    | 60000000000                                                                                                                                     | 60000000000                                                                                                                                     |
 |  max_runs        | -1                                                                                                                                              | -1                                                                                                                                              |
 |  time_unit       | ns                                                                                                                                              | ns                                                                                                                                              |
 |  using_scheduler | False                                                                                                                                           | False                                                                                                                                           |
 |  verify          | False                                                                                                                                           | False                                                                                                                                           |
 |  warmup_duration | 0                                                                                                                                               | 0                                                                                                                                               |
 +------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)   | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |      new |        ||      old |      new |        |         |
 +---------++----------+----------+--------++----------+----------+--------+---------+
+| 10a     ||    891.7 |    734.0 |  -18%  ||     1.12 |     1.36 |  +21%  |  0.0000 |
+| 10b     ||    868.3 |    719.0 |  -17%  ||     1.15 |     1.39 |  +21%  |  0.0000 |
+| 10c     ||   1752.0 |   1659.1 |   -5%  ||     0.57 |     0.60 |   +6%  |  0.0000 |
+| 11a     ||    314.3 |    270.6 |  -14%  ||     3.18 |     3.70 |  +16%  |  0.0000 |
+| 11b     ||    303.8 |    258.7 |  -15%  ||     3.29 |     3.87 |  +17%  |  0.0000 |
+| 11c     ||   1082.5 |    983.7 |   -9%  ||     0.92 |     1.02 |  +10%  |  0.0000 |
 | 11d     ||  11043.7 |  10678.2 |   -3%  ||     0.09 |     0.09 |   +3%  |       ˅ |
+| 12a     ||    628.6 |    588.5 |   -6%  ||     1.59 |     1.70 |   +7%  |  0.0002 |
+| 12b     ||    413.0 |    347.7 |  -16%  ||     2.42 |     2.88 |  +19%  |  0.0000 |
+| 12c     ||   7405.3 |   6655.4 |  -10%  ||     0.14 |     0.15 |  +11%  |       ˅ |
+| 13a     ||    361.6 |    330.5 |   -9%  ||     2.77 |     3.03 |   +9%  |  0.0000 |
+| 13b     ||    477.5 |    417.0 |  -13%  ||     2.09 |     2.40 |  +15%  |  0.0000 |
+| 13c     ||    387.0 |    315.4 |  -18%  ||     2.58 |     3.17 |  +23%  |  0.0000 |
+| 13d     ||    361.7 |    329.3 |   -9%  ||     2.76 |     3.04 |  +10%  |  0.0000 |
+| 14a     ||   8783.3 |   7993.7 |   -9%  ||     0.11 |     0.13 |  +10%  |       ˅ |
+| 14b     ||   9162.9 |   8368.4 |   -9%  ||     0.11 |     0.12 |   +9%  |       ˅ |
+| 14c     ||   8784.2 |   8009.9 |   -9%  ||     0.11 |     0.12 |  +10%  |       ˅ |
+| 15a     ||    430.5 |    400.2 |   -7%  ||     2.32 |     2.50 |   +8%  |  0.0000 |
+| 15b     ||    423.5 |    389.8 |   -8%  ||     2.36 |     2.57 |   +9%  |  0.0000 |
+| 15c     ||    338.8 |    304.9 |  -10%  ||     2.95 |     3.28 |  +11%  |  0.0000 |
 | 15d     ||    953.3 |    921.7 |   -3%  ||     1.05 |     1.08 |   +3%  |  0.0000 |
+| 16a     ||  10875.8 |  10196.9 |   -6%  ||     0.09 |     0.10 |   +7%  |       ˅ |
+| 16b     ||  14080.2 |  13235.3 |   -6%  ||     0.07 |     0.08 |   +6%  |       ˅ |
+| 16c     ||  11155.5 |  10555.8 |   -5%  ||     0.09 |     0.09 |   +6%  |       ˅ |
+| 16d     ||  11140.2 |  10450.7 |   -6%  ||     0.09 |     0.10 |   +7%  |       ˅ |
+| 17a     ||   2170.3 |   1883.3 |  -13%  ||     0.46 |     0.53 |  +15%  |  0.0000 |
+| 17b     ||   1793.4 |   1527.8 |  -15%  ||     0.56 |     0.65 |  +17%  |  0.0000 |
+| 17c     ||   1716.7 |   1454.7 |  -15%  ||     0.58 |     0.69 |  +18%  |  0.0000 |
+| 17d     ||   1973.2 |   1650.3 |  -16%  ||     0.51 |     0.61 |  +20%  |  0.0000 |
+| 17e     ||   6113.8 |   5418.7 |  -11%  ||     0.16 |     0.18 |  +13%  |  0.0000 |
+| 17f     ||   3411.3 |   2990.3 |  -12%  ||     0.29 |     0.33 |  +14%  |  0.0000 |
+| 18a     ||   1273.8 |   1180.3 |   -7%  ||     0.79 |     0.85 |   +8%  |  0.0000 |
+| 18b     ||    664.7 |    529.1 |  -20%  ||     1.50 |     1.89 |  +26%  |  0.0000 |
+| 18c     ||   7838.9 |   6995.1 |  -11%  ||     0.13 |     0.14 |  +12%  |       ˅ |
+| 19a     ||  11817.6 |  10489.6 |  -11%  ||     0.08 |     0.10 |  +13%  |       ˅ |
+| 19b     ||   6147.0 |   4915.3 |  -20%  ||     0.16 |     0.20 |  +25%  |  0.0000 |
+| 19c     ||  11491.2 |  10164.3 |  -12%  ||     0.09 |     0.10 |  +13%  |       ˅ |
 | 19d     ||  11178.6 |  10756.9 |   -4%  ||     0.09 |     0.09 |   +4%  |       ˅ |
+| 1a      ||    138.1 |    114.1 |  -17%  ||     7.24 |     8.76 |  +21%  |  0.0000 |
+| 1b      ||     58.3 |     48.7 |  -17%  ||    17.14 |    20.55 |  +20%  |  0.0000 |
+| 1c      ||     58.5 |     48.9 |  -16%  ||    17.09 |    20.46 |  +20%  |  0.0000 |
+| 1d      ||     58.2 |     48.6 |  -16%  ||    17.18 |    20.57 |  +20%  |  0.0000 |
+| 20a     ||   2731.7 |   2579.0 |   -6%  ||     0.37 |     0.39 |   +6%  |  0.0115 |
 | 20b     ||   2315.1 |   2334.7 |   +1%  ||     0.43 |     0.43 |   -1%  |  0.0128 |
 | 20c     ||   1431.4 |   1441.2 |   +1%  ||     0.70 |     0.69 |   -1%  |  0.0491 |
 | 21a     ||   8105.0 |   8358.0 |   +3%  ||     0.12 |     0.12 |   -3%  |       ˅ |
 | 21b     ||    468.3 |    469.4 |   +0%  ||     2.14 |     2.13 |   -0%  |  0.6174 |
 | 21c     ||   8423.7 |   8503.7 |   +1%  ||     0.12 |     0.12 |   -1%  |       ˅ |
 | 22a     ||   7055.9 |   7128.2 |   +1%  ||     0.14 |     0.14 |   -1%  |       ˅ |
 | 22b     ||   7060.7 |   7107.9 |   +1%  ||     0.14 |     0.14 |   -1%  |       ˅ |
 | 22c     ||   8788.0 |   8789.0 |   +0%  ||     0.11 |     0.11 |   -0%  |       ˅ |
 | 22d     ||   8829.2 |   8822.9 |   -0%  ||     0.11 |     0.11 |   +0%  |       ˅ |
 | 23a     ||    338.3 |    335.3 |   -1%  ||     2.96 |     2.98 |   +1%  |  0.2047 |
 | 23b     ||    255.7 |    256.4 |   +0%  ||     3.91 |     3.90 |   -0%  |  0.6501 |
 | 23c     ||    360.4 |    356.2 |   -1%  ||     2.77 |     2.81 |   +1%  |  0.0989 |
+| 24a     ||   6362.5 |   5680.2 |  -11%  ||     0.16 |     0.18 |  +12%  |  0.0000 |
+| 24b     ||   6239.9 |   5520.5 |  -12%  ||     0.16 |     0.18 |  +13%  |  0.0001 |
 | 25a     ||    675.4 |    668.8 |   -1%  ||     1.48 |     1.50 |   +1%  |  0.1400 |
 | 25b     ||    702.8 |    697.3 |   -1%  ||     1.42 |     1.43 |   +1%  |  0.1628 |
 | 25c     ||   7850.3 |   7904.5 |   +1%  ||     0.13 |     0.13 |   -1%  |       ˅ |
 | 26a     ||   1220.4 |   1210.5 |   -1%  ||     0.82 |     0.83 |   +1%  |  0.3545 |
 | 26b     ||   1190.1 |   1182.6 |   -1%  ||     0.84 |     0.85 |   +1%  |  0.4666 |
 | 26c     ||   1219.4 |   1209.7 |   -1%  ||     0.82 |     0.83 |   +1%  |  0.3699 |
 | 27a     ||   7004.8 |   7168.5 |   +2%  ||     0.14 |     0.14 |   -2%  |       ˅ |
 | 27b     ||   6957.7 |   7095.6 |   +2%  ||     0.14 |     0.14 |   -2%  |       ˅ |
 | 27c     ||    430.2 |    430.2 |   +0%  ||     2.32 |     2.32 |   -0%  |  0.9942 |
 | 28a     ||   8943.1 |   8935.4 |   -0%  ||     0.11 |     0.11 |   +0%  |       ˅ |
 | 28b     ||   6940.7 |   7065.2 |   +2%  ||     0.14 |     0.14 |   -2%  |       ˅ |
 | 28c     ||   8939.8 |   8928.8 |   -0%  ||     0.11 |     0.11 |   +0%  |       ˅ |
+| 29a     ||   6267.4 |   5481.0 |  -13%  ||     0.16 |     0.18 |  +14%  |  0.2042 |
 | 29b     ||    388.9 |    388.1 |   -0%  ||     2.57 |     2.58 |   +0%  |  0.9844 |
+| 29c     ||   6681.2 |   5838.2 |  -13%  ||     0.15 |     0.17 |  +14%  |       ˅ |
 | 2a      ||    205.4 |    208.8 |   +2%  ||     4.87 |     4.79 |   -2%  |  0.0000 |
 | 2b      ||    172.2 |    171.9 |   -0%  ||     5.81 |     5.82 |   +0%  |  0.0888 |
 | 2c      ||    120.5 |    120.0 |   -0%  ||     8.30 |     8.33 |   +0%  |  0.0000 |
 | 2d      ||    255.7 |    254.5 |   -0%  ||     3.91 |     3.93 |   +0%  |  0.0000 |
 | 30a     ||    712.8 |    705.6 |   -1%  ||     1.40 |     1.42 |   +1%  |  0.7507 |
 | 30b     ||   1089.5 |   1071.7 |   -2%  ||     0.92 |     0.93 |   +2%  |  0.6161 |
 | 30c     ||   7968.5 |   8158.7 |   +2%  ||     0.13 |     0.12 |   -2%  |       ˅ |
 | 31a     ||    789.7 |    783.8 |   -1%  ||     1.27 |     1.28 |   +1%  |  0.6653 |
 | 31b     ||   1161.1 |   1145.0 |   -1%  ||     0.86 |     0.87 |   +1%  |  0.4692 |
 | 31c     ||    798.4 |    795.7 |   -0%  ||     1.25 |     1.26 |   +0%  |  0.8417 |
 | 32a     ||    106.5 |    106.0 |   -0%  ||     9.39 |     9.43 |   +0%  |  0.0000 |
 | 32b     ||    550.5 |    552.7 |   +0%  ||     1.82 |     1.81 |   -0%  |  0.0049 |
 | 33a     ||    169.6 |    170.1 |   +0%  ||     5.89 |     5.88 |   -0%  |  0.7984 |
 | 33b     ||    168.2 |    168.6 |   +0%  ||     5.94 |     5.93 |   -0%  |  0.8465 |
 | 33c     ||    192.9 |    193.1 |   +0%  ||     5.18 |     5.18 |   -0%  |  0.9054 |
 | 3a      ||   8292.1 |   8397.9 |   +1%  ||     0.12 |     0.12 |   -1%  |       ˅ |
 | 3b      ||     80.3 |     80.3 |   -0%  ||    12.45 |    12.45 |   +0%  |  0.7112 |
 | 3c      ||  10477.9 |  10522.2 |   +0%  ||     0.10 |     0.10 |   -0%  |       ˅ |
 | 4a      ||    264.9 |    263.4 |   -1%  ||     3.78 |     3.80 |   +1%  |  0.0013 |
 | 4b      ||     63.4 |     61.8 |   -3%  ||    15.77 |    16.18 |   +3%  |  0.0000 |
 | 4c      ||    300.6 |    298.9 |   -1%  ||     3.33 |     3.35 |   +1%  |  0.0160 |
 | 5a      ||   7975.3 |   8070.5 |   +1%  ||     0.13 |     0.12 |   -1%  |       ˅ |
 | 5b      ||    523.4 |    526.6 |   +1%  ||     1.91 |     1.90 |   -1%  |  0.0123 |
 | 5c      ||   9233.3 |   9256.6 |   +0%  ||     0.11 |     0.11 |   -0%  |       ˅ |
 | 6a      ||    662.0 |    665.9 |   +1%  ||     1.51 |     1.50 |   -1%  |  0.0000 |
 | 6b      ||   2281.7 |   2269.0 |   -1%  ||     0.44 |     0.44 |   +1%  |  0.0970 |
 | 6c      ||    661.6 |    662.5 |   +0%  ||     1.51 |     1.51 |   -0%  |  0.1938 |
 | 6d      ||   2275.5 |   2273.6 |   -0%  ||     0.44 |     0.44 |   +0%  |  0.7848 |
 | 6e      ||    664.3 |    665.0 |   +0%  ||     1.51 |     1.50 |   -0%  |  0.2515 |
 | 6f      ||   3346.0 |   3311.5 |   -1%  ||     0.30 |     0.30 |   +1%  |  0.0001 |
 | 7a      ||    463.8 |    461.0 |   -1%  ||     2.16 |     2.17 |   +1%  |  0.7640 |
 | 7b      ||    283.2 |    274.4 |   -3%  ||     3.53 |     3.64 |   +3%  |  0.1120 |
 | 7c      ||  23325.6 |  23593.9 |   +1%  ||     0.04 |     0.04 |   -1%  |       ˅ |
+| 8a      ||    240.9 |    197.4 |  -18%  ||     4.15 |     5.07 |  +22%  |  0.0000 |
+| 8b      ||    338.4 |    277.8 |  -18%  ||     2.96 |     3.60 |  +22%  |  0.0000 |
+| 8c      ||   8043.1 |   7596.6 |   -6%  ||     0.12 |     0.13 |   +6%  |       ˅ |
+| 8d      ||   2391.2 |   2272.8 |   -5%  ||     0.42 |     0.44 |   +5%  |  0.0000 |
 | 9a      ||   6467.6 |   6205.5 |   -4%  ||     0.15 |     0.16 |   +4%  |  0.0004 |
+| 9b      ||    646.5 |    573.4 |  -11%  ||     1.55 |     1.74 |  +13%  |  0.0000 |
 | 9c      ||   6643.4 |   6370.9 |   -4%  ||     0.15 |     0.16 |   +4%  |  0.0002 |
 | 9d      ||   7919.2 |   7587.4 |   -4%  ||     0.13 |     0.13 |   +4%  |       ˅ |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | Sum     || 422827.4 | 404560.0 |   -4%  ||          |          |        |         |
+| Geomean ||          |          |        ||          |          |   +6%  |         |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 |         || ˅ Insufficient number of runs for p-value calculation                  |
 +---------++----------+----------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkJoinOrder - multi-threaded (50 clients)**
<details>
<summary>
Sum of avg. item runtimes: -1%
 || 
Geometric mean of throughput changes: +0%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+-------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                     | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkJoinOrder_e1389f276e4e0fbfec34eca098365999553c05f8_mt.json | /home/Julian.Menzler/repo_hyrise2/cmake_release/benchmark_all_results/hyriseBenchmarkJoinOrder_fd52afb2fff07d264326398bac7f39ccaad522e5_mt.json |
 +-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                     | e1389f276e4e0fbfec34eca098365999553c05f8-dirty                                                                                                  | fd52afb2fff07d264326398bac7f39ccaad522e5-dirty                                                                                                  |
 |  benchmark_mode               | Ordered                                                                                                                                         | Ordered                                                                                                                                         |
 |  build_type                   | release                                                                                                                                         | release                                                                                                                                         |
 |  chunk_size                   | 65535                                                                                                                                           | 65535                                                                                                                                           |
 |  clients                      | 50                                                                                                                                              | 50                                                                                                                                              |
 |  compiler                     | gcc 9.2                                                                                                                                         | gcc 9.2                                                                                                                                         |
 |  cores                        | 0                                                                                                                                               | 0                                                                                                                                               |
 |  date                         | 2021-05-11 15:24:44                                                                                                                             | 2021-05-11 23:17:41                                                                                                                             |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                                                                         | {'default': {'encoding': 'Dictionary'}}                                                                                                         |
 |  indexes                      | False                                                                                                                                           | False                                                                                                                                           |
 |  max_duration                 | 60000000000                                                                                                                                     | 60000000000                                                                                                                                     |
 |  max_runs                     | -1                                                                                                                                              | -1                                                                                                                                              |
 |  time_unit                    | ns                                                                                                                                              | ns                                                                                                                                              |
 |  using_scheduler              | True                                                                                                                                            | True                                                                                                                                            |
 |  utilized_cores_per_numa_node | [20, 20, 20, 20]                                                                                                                                | [20, 20, 20, 20]                                                                                                                                |
 |  verify                       | False                                                                                                                                           | False                                                                                                                                           |
 |  warmup_duration              | 0                                                                                                                                               | 0                                                                                                                                               |
 +-------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)   | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |      new |        ||      old |      new |        |         |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | 10a     ||   1600.2 |   1555.4 |   -3%  ||    29.86 |    30.62 |   +3%  |  0.4894 |
 | 10b     ||   1569.7 |   1570.3 |   +0%  ||    30.90 |    30.53 |   -1%  |  0.9928 |
 | 10c     ||   2781.0 |   2803.1 |   +1%  ||    16.67 |    16.93 |   +2%  |  0.8407 |
-| 11a     ||    364.9 |    437.0 |  +20%  ||   134.34 |   111.70 |  -17%  |  0.0000 |
-| 11b     ||    336.3 |    407.2 |  +21%  ||   145.57 |   119.57 |  -18%  |  0.0000 |
 | 11c     ||   1845.4 |   1822.8 |   -1%  ||    26.32 |    26.61 |   +1%  |  0.5966 |
+| 11d     ||  25320.8 |  25863.8 |   +2%  ||     1.37 |     1.55 |  +13%  |  0.6572 |
 | 12a     ||   1130.0 |   1106.5 |   -2%  ||    43.30 |    44.34 |   +2%  |  0.3922 |
 | 12b     ||   1526.0 |   1512.9 |   -1%  ||    31.40 |    31.30 |   -0%  |  0.8521 |
 | 12c     ||   9739.3 |   9844.9 |   +1%  ||     4.07 |     4.07 |   -0%  |  0.8824 |
 | 13a     ||    653.7 |    661.8 |   +1%  ||    74.91 |    74.26 |   -1%  |  0.6007 |
 | 13b     ||   1631.5 |   1614.9 |   -1%  ||    29.27 |    29.51 |   +1%  |  0.8305 |
 | 13c     ||   1445.1 |   1435.3 |   -1%  ||    32.86 |    33.13 |   +1%  |  0.8689 |
 | 13d     ||    649.8 |    657.3 |   +1%  ||    75.20 |    74.21 |   -1%  |  0.6705 |
 | 14a     ||   9794.6 |  10440.2 |   +7%  ||     3.97 |     4.02 |   +1%  |  0.4049 |
 | 14b     ||  11307.8 |  10481.3 |   -7%  ||     3.63 |     3.62 |   -0%  |  0.2596 |
 | 14c     ||   9959.2 |   9611.7 |   -3%  ||     3.98 |     3.98 |   -0%  |  0.6397 |
 | 15a     ||    557.0 |    553.7 |   -1%  ||    88.08 |    88.50 |   +0%  |  0.7698 |
 | 15b     ||    537.8 |    530.5 |   -1%  ||    91.24 |    92.40 |   +1%  |  0.4763 |
 | 15c     ||    393.9 |    393.2 |   -0%  ||   124.53 |   124.54 |   +0%  |  0.9256 |
 | 15d     ||   1603.7 |   1552.5 |   -3%  ||    30.37 |    31.33 |   +3%  |  0.1772 |
+| 16a     ||  18719.4 |  17646.1 |   -6%  ||     1.85 |     2.00 |   +8%  |  0.3007 |
 | 16b     ||  26343.9 |  24587.5 |   -7%  ||     1.35 |     1.33 |   -1%  |  0.2035 |
 | 16c     ||  19687.7 |  18394.2 |   -7%  ||     1.93 |     2.00 |   +3%  |  0.2348 |
 | 16d     ||  19687.3 |  20111.9 |   +2%  ||     1.92 |     1.98 |   +3%  |  0.7486 |
 | 17a     ||   3286.1 |   3187.4 |   -3%  ||    14.22 |    14.60 |   +3%  |  0.5630 |
 | 17b     ||   2621.0 |   2444.7 |   -7%  ||    17.58 |    17.43 |   -1%  |  0.1808 |
 | 17c     ||   2536.3 |   2523.3 |   -1%  ||    18.40 |    18.40 |   -0%  |  0.9304 |
 | 17d     ||   2762.7 |   2696.2 |   -2%  ||    16.68 |    16.55 |   -1%  |  0.6528 |
 | 17e     ||   8669.4 |   8271.8 |   -5%  ||     5.33 |     5.48 |   +3%  |  0.2709 |
 | 17f     ||   5318.2 |   5187.2 |   -2%  ||     8.65 |     8.73 |   +1%  |  0.6611 |
 | 18a     ||   1790.4 |   1750.5 |   -2%  ||    27.02 |    26.93 |   -0%  |  0.5256 |
 | 18b     ||   1574.8 |   1573.8 |   -0%  ||    30.83 |    30.45 |   -1%  |  0.9885 |
 | 18c     ||   9183.5 |   9376.4 |   +2%  ||     4.10 |     4.18 |   +2%  |  0.7866 |
+| 19a     ||  12287.3 |  13068.2 |   +6%  ||     2.70 |     2.92 |   +8%  |  0.5173 |
+| 19b     ||   5368.4 |   5725.8 |   +7%  ||     6.92 |     7.48 |   +8%  |  0.4552 |
 | 19c     ||  11665.9 |  11668.4 |   +0%  ||     2.88 |     2.93 |   +2%  |  0.9981 |
 | 19d     ||  16636.2 |  17498.1 |   +5%  ||     2.03 |     2.12 |   +4%  |  0.3482 |
 | 1a      ||    266.7 |    266.9 |   +0%  ||   181.80 |   182.16 |   +0%  |  0.9594 |
 | 1b      ||     61.6 |     61.2 |   -1%  ||   743.08 |   746.91 |   +1%  |  0.3694 |
 | 1c      ||     62.5 |     62.5 |   +0%  ||   733.34 |   732.71 |   -0%  |  0.8978 |
 | 1d      ||     61.6 |     61.2 |   -1%  ||   742.53 |   746.75 |   +1%  |  0.2817 |
+| 20a     ||   4076.6 |   3914.1 |   -4%  ||    10.88 |    11.43 |   +5%  |  0.5040 |
 | 20b     ||   3674.1 |   3802.2 |   +3%  ||    11.68 |    12.17 |   +4%  |  0.6069 |
 | 20c     ||   2871.7 |   2822.8 |   -2%  ||    16.67 |    16.63 |   -0%  |  0.7606 |
 | 21a     ||   8773.2 |   9372.9 |   +7%  ||     4.42 |     4.27 |   -3%  |  0.4221 |
-| 21b     ||    615.8 |    732.7 |  +19%  ||    79.41 |    65.84 |  -17%  |  0.0000 |
 | 21c     ||   8547.5 |   9163.4 |   +7%  ||     4.10 |     4.08 |   -0%  |  0.4589 |
 | 22a     ||   8879.3 |   9427.8 |   +6%  ||     4.33 |     4.43 |   +2%  |  0.4221 |
 | 22b     ||   9135.3 |   8538.3 |   -7%  ||     4.33 |     4.35 |   +0%  |  0.3553 |
 | 22c     ||   9047.4 |   8881.8 |   -2%  ||     3.92 |     3.93 |   +0%  |  0.7914 |
 | 22d     ||  10375.1 |   9312.4 |  -10%  ||     3.93 |     3.92 |   -0%  |  0.1500 |
 | 23a     ||    385.5 |    388.2 |   +1%  ||   126.60 |   126.07 |   -0%  |  0.7578 |
 | 23b     ||    515.6 |    518.6 |   +1%  ||    95.05 |    94.69 |   -0%  |  0.7975 |
 | 23c     ||    433.8 |    433.1 |   -0%  ||   112.70 |   112.70 |   -0%  |  0.9382 |
+| 24a     ||   6335.8 |   6124.2 |   -3%  ||     6.55 |     6.90 |   +5%  |  0.6519 |
+| 24b     ||   6336.0 |   5945.9 |   -6%  ||     6.70 |     7.17 |   +7%  |  0.3241 |
 | 25a     ||   1553.7 |   1524.0 |   -2%  ||    31.23 |    31.31 |   +0%  |  0.6196 |
 | 25b     ||   1672.8 |   1659.7 |   -1%  ||    28.90 |    28.98 |   +0%  |  0.8565 |
 | 25c     ||   9604.4 |   9223.8 |   -4%  ||     4.10 |     4.13 |   +1%  |  0.5552 |
 | 26a     ||   2648.1 |   2602.3 |   -2%  ||    17.57 |    18.08 |   +3%  |  0.7512 |
 | 26b     ||   2600.2 |   2564.4 |   -1%  ||    18.00 |    18.23 |   +1%  |  0.7986 |
 | 26c     ||   2628.0 |   2643.9 |   +1%  ||    17.56 |    17.68 |   +1%  |  0.9113 |
 | 27a     ||   8345.1 |   7594.2 |   -9%  ||     4.45 |     4.33 |   -3%  |  0.2049 |
 | 27b     ||   8266.7 |   8901.0 |   +8%  ||     4.47 |     4.52 |   +1%  |  0.3771 |
-| 27c     ||    619.3 |    781.7 |  +26%  ||    79.00 |    62.54 |  -21%  |  0.0000 |
 | 28a     ||   9585.1 |   9387.0 |   -2%  ||     3.80 |     3.83 |   +1%  |  0.7687 |
 | 28b     ||   9114.7 |   8541.6 |   -6%  ||     4.38 |     4.40 |   +0%  |  0.4068 |
 | 28c     ||  10148.5 |   9551.8 |   -6%  ||     3.85 |     3.82 |   -1%  |  0.4074 |
 | 29a     ||   6736.7 |   5972.8 |  -11%  ||     6.27 |     6.37 |   +2%  |  0.1184 |
 | 29b     ||    838.7 |    836.4 |   -0%  ||    58.03 |    58.28 |   +0%  |  0.9403 |
+| 29c     ||   6934.4 |   6440.4 |   -7%  ||     6.00 |     6.32 |   +5%  |  0.3209 |
 | 2a      ||    337.7 |    342.0 |   +1%  ||   145.03 |   143.43 |   -1%  |  0.4691 |
 | 2b      ||    287.9 |    288.9 |   +0%  ||   169.78 |   169.53 |   -0%  |  0.7943 |
 | 2c      ||    181.9 |    183.6 |   +1%  ||   266.07 |   263.63 |   -1%  |  0.4581 |
 | 2d      ||    414.8 |    410.3 |   -1%  ||   118.37 |   119.45 |   +1%  |  0.5261 |
 | 30a     ||   1542.8 |   1601.2 |   +4%  ||    29.43 |    29.76 |   +1%  |  0.4399 |
 | 30b     ||   2653.6 |   2739.5 |   +3%  ||    16.45 |    16.60 |   +1%  |  0.5671 |
 | 30c     ||  10221.4 |   9769.9 |   -4%  ||     3.93 |     4.02 |   +2%  |  0.5500 |
 | 31a     ||   1772.1 |   1807.4 |   +2%  ||    26.82 |    26.63 |   -1%  |  0.6752 |
 | 31b     ||   3059.4 |   3058.8 |   -0%  ||    15.37 |    15.36 |   -0%  |  0.9975 |
 | 31c     ||   1807.6 |   1828.1 |   +1%  ||    26.63 |    26.57 |   -0%  |  0.7842 |
 | 32a     ||    115.6 |    117.1 |   +1%  ||   411.89 |   406.44 |   -1%  |  0.1619 |
 | 32b     ||    883.9 |    882.3 |   -0%  ||    55.76 |    55.81 |   +0%  |  0.9040 |
-| 33a     ||    249.5 |    271.3 |   +9%  ||   195.36 |   179.67 |   -8%  |  0.0000 |
-| 33b     ||    246.6 |    269.7 |   +9%  ||   197.72 |   181.13 |   -8%  |  0.0000 |
-| 33c     ||    302.2 |    321.1 |   +6%  ||   161.98 |   152.38 |   -6%  |  0.0000 |
 | 3a      ||   9100.6 |   9614.6 |   +6%  ||     4.17 |     4.10 |   -2%  |  0.4429 |
 | 3b      ||    133.0 |    131.4 |   -1%  ||   360.86 |   365.59 |   +1%  |  0.1689 |
 | 3c      ||  11088.5 |  11170.6 |   +1%  ||     3.28 |     3.30 |   +1%  |  0.9251 |
 | 4a      ||    440.5 |    431.1 |   -2%  ||   111.59 |   113.50 |   +2%  |  0.1834 |
 | 4b      ||     69.0 |     69.9 |   +1%  ||   667.35 |   659.46 |   -1%  |  0.0514 |
 | 4c      ||    512.9 |    508.2 |   -1%  ||    95.63 |    96.46 |   +1%  |  0.5879 |
 | 5a      ||   9403.8 |   9486.8 |   +1%  ||     4.32 |     4.30 |   -0%  |  0.9111 |
 | 5b      ||    674.2 |    666.9 |   -1%  ||    72.77 |    73.65 |   +1%  |  0.6947 |
 | 5c      ||  10805.3 |  10761.5 |   -0%  ||     3.75 |     3.72 |   -1%  |  0.9576 |
 | 6a      ||   1569.3 |   1579.1 |   +1%  ||    29.85 |    30.10 |   +1%  |  0.9021 |
 | 6b      ||   3108.4 |   2986.4 |   -4%  ||    15.27 |    15.48 |   +1%  |  0.4695 |
 | 6c      ||   1623.3 |   1608.6 |   -1%  ||    29.66 |    30.16 |   +2%  |  0.8441 |
 | 6d      ||   2899.5 |   3003.1 |   +4%  ||    15.50 |    15.61 |   +1%  |  0.4709 |
 | 6e      ||   1557.2 |   1603.9 |   +3%  ||    29.80 |    29.95 |   +1%  |  0.5528 |
 | 6f      ||   4834.4 |   4713.7 |   -2%  ||     9.65 |     9.92 |   +3%  |  0.5495 |
 | 7a      ||    645.3 |    664.2 |   +3%  ||    75.97 |    73.53 |   -3%  |  0.2321 |
-| 7b      ||    443.0 |    468.7 |   +6%  ||   110.20 |   104.35 |   -5%  |  0.0041 |
 | 7c      ||  45457.8 |  47611.7 |   +5%  ||     0.75 |     0.77 |   +2%  |  0.0033 |
 | 8a      ||    511.2 |    519.4 |   +2%  ||    94.78 |    93.64 |   -1%  |  0.4809 |
+| 8b      ||    560.6 |    529.5 |   -6%  ||    87.28 |    92.19 |   +6%  |  0.0323 |
 | 8c      ||  12183.1 |  12065.6 |   -1%  ||     3.42 |     3.48 |   +2%  |  0.8707 |
 | 8d      ||   3719.5 |   3618.8 |   -3%  ||    12.77 |    13.07 |   +2%  |  0.5637 |
 | 9a      ||   7875.7 |   8420.2 |   +7%  ||     4.68 |     4.65 |   -1%  |  0.3455 |
 | 9b      ||   1164.5 |   1156.2 |   -1%  ||    41.86 |    42.00 |   +0%  |  0.8499 |
 | 9c      ||   8715.3 |   8540.0 |   -2%  ||     4.52 |     4.55 |   +1%  |  0.7761 |
 | 9d      ||  10650.7 |  10894.6 |   +2%  ||     3.52 |     3.67 |   +4%  |  0.6943 |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | Sum     || 588461.3 | 585339.3 |   -1%  ||          |          |        |         |
 | Geomean ||          |          |        ||          |          |   +0%  |         |
 +---------++----------+----------+--------++----------+----------+--------+---------+
```
</details>